daemon/daemon.go

```diff
@@ -353,17 +353,43 @@ func (daemon *Daemon) restore() error {
 				// The container register failed should not be started.
 				return
 			}
+		}(c.container, c.registered)
```

Comment:
Hmmm, you are probably right...
Maybe restarting the daemon after unclean shutdown would be a bit slower (making a few syscalls to cleanup)...